firedrake/mg/embedded.py

```diff
@@ -56,14 +56,15 @@ def __new__(cls, element, use_fortin_interpolation=True):
     def __init__(self, element, use_fortin_interpolation=True):
         degree = element.degree()
         cell = element.cell()
+        family = "DQ" if cell in (ufl.quadrilateral, ufl.hexahedron) else "DG"
```

Comment:
Does this actually do the right thing in 3D? Won't the cell in 3D be a tensor product cell? I suspect slightly more sophisticated logic is needed.

Reply:
OK, I've restricted these changes to 2D. (I'm not familiar enough with the tensor product stuff in firedrake to make the right changes, as I've never used it.)